packages/cli/lib/ModuleManager.ts

```diff
@@ -0,0 +1,221 @@
+      throw new Error(pluginNotFound(name, type));
+    }
+
+    return this._plugins.get(type).get(name);
+  }
+
+  getPluginsOfType(type: string) {
+    if (!this._plugins.has(type)) {
+      throw new Error(pluginsNotFound(type));
+    }
+
+    return this._plugins.get(type);
+  }
+
+  async prepare() {
+    for (const application of this._applications.values()) {
+      if (application.prepare) {
+        await application.prepare();
+      }
+    }
+
+    for (const plugins of this.plugins.values()) {
+      for (const plugin of plugins.values()) {
+        if (plugin.prepare) {
+          await plugin.prepare();
+        }
+      }
+    }
+  }
+
+  async cleanup() {
+    for (const application of this._applications.values()) {
+      if (application.cleanup) {
+        await application.cleanup();
+      }
+    }
+
+    for (const plugins of this.plugins.values()) {
+      for (const plugin of plugins.values()) {
+        if (plugin.cleanup) {
+          await plugin.cleanup();
+        }
+      }
+    }
+  }
+
+  async configureApplicationCommands(yargs: Yargs.Argv) {
+    for (const application of this._applications.values()) {
+      const commandModules = await application.getCommandModules();
+      for (const commandModule of commandModules) {
+        // check requirements
+        if (!commandModule.command) {
+          throw new Error("SynTest command module required command property!");
+        } else if (!commandModule.describe) {
+          throw new Error("SynTest command module required describe property!");
+        } else if (!commandModule.handler) {
+          throw new Error("SynTest command module required handler function!");
+        }
+        yargs = yargs.command(commandModule);
+      }
+    }
+
+    return yargs;
+  }
+
+  async configurePluginOptions(yargs: Yargs.Argv) {
+    for (const plugins of this.plugins.values()) {
+      for (const plugin of plugins.values()) {
+        if (plugin.getOptions) {
+          const options = await plugin.getOptions();
+
+          for (const option of options.keys()) {
+            yargs = yargs.option(`${plugin.name}-${option}`, {
+              ...options.get(option),
+              group: `${plugin.name} ${options.get(option).group || ""}`,
+            });
+          }
+        }
+      }
+    }
+    return yargs;
+  }
+
+  async getModulePath(module: string): Promise<string> {
+    let modulePath = "";
+
+    if (module.startsWith("file:")) {
+      // It is a file path
+      modulePath = path.resolve(module.replace("file:", ""));
+      if (!shell.test("-e", modulePath)) {
+        throw new Error(`Filepath does not lead to an module, path: ${module}`);
+      }
+    } else {
+      // It is a npm package
+      modulePath = path.resolve(path.join("node_modules", module));
+
+      if (!shell.test("-e", modulePath)) {
+        // it is not locally installed lets try global
+        modulePath = path.resolve(path.join(globalModules, module));
```

Comment:
Do we need the path, can't we just import the package?

Reply:
I tried but I can't get it to work, the dynamic import seems to use the path relative to the ModuleManager.ts file instead of the current working directory.

Reply:
Okay. Maybe we could make some upward-looking util function that can be used to look for a certain file or directory in the current directory and search parent directories if that file or folder is not found. Currently, the tool would not work when run from a subfolder.

Reply:
I created an issue for this.